ibc-core/ics02-client/context/src/client_state.rs

```diff
@@ -124,6 +124,20 @@ where
 
     /// Returns the status of the client. Only Active clients are allowed to process packets.
     fn status(&self, ctx: &V, client_id: &ClientId) -> Result<Status, ClientError>;
+
+    /// Verifies whether the calling (subject) client state matches the substitute
+    /// client state for the purposes of client recovery.
+    ///
+    /// Note that this validation function does not need to perform *all* of the
+    /// validation steps necessary to confirm client recovery. Some checks, such
+    /// as checking that the subject client state's latest height < the substitute
+    /// client's latest height, as well as checking that the subject client is
+    /// inactive and that the substitute client is active, are performed by the
+    /// `validate` function in the `recover_client` module at the ics02-client
+    /// level.
+    ///
+    /// Returns `Ok` if the subject and substitute client states match, `Err` otherwise.
+    fn check_substitute(&self, ctx: &V, substitute_client_state: Any) -> Result<(), ClientError>;
```

Comment:
do we need `substitute_client_state: Any`? can it be `substitute_client_state: Self`?

Reply:
`Any` is used pretty consistently throughout the codebase instead of `Self`; I'm going to stick with that for now.